### PR TITLE
Add executable to observe Kerr Schild

### PR DESCRIPTION
**Pull request by geoffrey4444**
## Proposed changes

This is a prototype executable to evolve KerrSchild. Generalized Harmonic isn't finished yet, but the executable does much of what the final one will need: using the system GeneralizedHarmonic, load Kerr-Schild initial data onto a grid while observing the solution at several times. As an example, the executable observes the SpatialRicciScalar and Lapse, which are the ingredients for creating pseudo-embedding diagrams.

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by geoffrey4444** [comment]
No longer uses a hack to observe Kerr Schild while loading a ScalarWave solution. The executable now uses the GeneralizedHarmonic system, loads analytic initial data from KerrSchild onto the grid, and then observes the analytic solution.

This PR is now a prototype for an executable to numerically evolve Kerr Schild, comparing to an analytic solution.

File: src/Evolution/Executables/GeneralizedHarmonic/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    LinearOperators
+    MathFunctions
+    Time
+    Utilities
+    ${SPECTRE_LIBRARIES}
+  )
+
+add_spectre_executable(
+  EvolveKerrSchild
+  EvolveKerrSchild
```
Comment (moxcodes) [inline src/Evolution/Executables/GeneralizedHarmonic/CMakeLists.txt:22]: There are two?

File: src/Evolution/Executables/GeneralizedHarmonic/EvolveKerrSchild.hpp [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Actions/ComputeNonconservativeBoundaryFluxes.hpp"  // IWYU pragma: keep
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Actions/FluxCommunication.hpp"  // IWYU pragma: keep
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.tpp"
+#include "Options/Options.hpp"
+#include "Parallel/InitializationFunctions.hpp"
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/WaveEquation/PlaneWave.hpp"  // IWYU pragma: keep
```
Comment (moxcodes) [inline src/Evolution/Executables/GeneralizedHarmonic/EvolveKerrSchild.hpp:43]: remove plane wave include

File: src/Evolution/Executables/GeneralizedHarmonic/EvolveKerrSchild.hpp [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+#include "Evolution/Systems/GeneralizedHarmonic/System.hpp"
+#include "Evolution/Systems/GeneralizedHarmonic/Tags.hpp"
+#include "IO/Observer/Actions.hpp"
+#include "IO/Observer/ArrayComponentId.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ObserverComponent.hpp"
+#include "IO/Observer/VolumeActions.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Actions/ApplyBoundaryFluxesGlobalTimeStepping.hpp"  // IWYU pragma: keep
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Actions/ComputeNonconservativeBoundaryFluxes.hpp"  // IWYU pragma: keep
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Actions/FluxCommunication.hpp"  // IWYU pragma: keep
```
Comment (moxcodes) [inline src/Evolution/Executables/GeneralizedHarmonic/EvolveKerrSchild.hpp:35]: remove flux stuff

File: src/Evolution/Executables/GeneralizedHarmonic/EvolveKerrSchild.hpp [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+  template <typename... InboxTags, typename Metavariables, typename ActionList,
+            typename ParallelComponent>
+  static auto apply(const db::DataBox<tmpl::list<>>& /*box*/,
+                    const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ElementIndex<3>& array_index,
+                    const ActionList /*meta*/,
+                    const ParallelComponent* const /*meta*/,
+                    std::vector<std::array<size_t, 3>> initial_extents,
+                    Domain<3, Frame::Inertial> domain,
```
Comment (moxcodes) [inline src/Evolution/Executables/GeneralizedHarmonic/EvolveKerrSchild.hpp:230]: Try rvalue reference (`Domain<3, Frame::Inertial>&& domain`) and a `std::move` when used

File: src/Evolution/Executables/GeneralizedHarmonic/EvolveKerrSchild.hpp [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+    // Compute the Ricci scalar with a numerical derivative of the
+    // Christoffel symbols
+    const auto& deriv_spatial_metric =
+        get<Tags::deriv<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>,
+                        tmpl::size_t<3>, Frame::Inertial>>(variables);
+    const auto& spatial_metric =
+        get<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>>(variables);
+    const auto& inverse_spatial_metric =
+        determinant_and_inverse(spatial_metric).second;
+    const auto& christoffel_second_kind = raise_or_lower_first_index(
```
Comment (moxcodes) [inline src/Evolution/Executables/GeneralizedHarmonic/EvolveKerrSchild.hpp:283]: Make variables earlier to avoid the copy

**Review by moxcodes — COMMENTED** [review]
group code review with @markscheel @mgiesler @geoffrey4444

**Comment by geoffrey4444** [comment]
Test times out when debugging is enabled

**Comment by nilsdeppe** [comment]
Marking this as "in progress" for now since we aren't sure whether we want it in the code base and since it might evolve into the actual single black hole executable

**Comment by nilsdeppe** [comment]
Since there seems to be no plan to ever merge this, can we close it?

**Comment by geoffrey4444** [comment]
Superseded by #1358

**Pull request closed without merging**